test/identify/service.node.ts

```diff
@@ -0,0 +1,163 @@
+      throw new Error('Identity service was not configured')
+    }
+
+    const identityServiceIdentifySpy = sinon.spy(libp2p.identifyService, 'identify')
+
+    // dial local -> remote via loopback in order to assert we receive the announce address via identify
+    const connection = await libp2p.dial(multiaddr(`/ip4/127.0.0.1/tcp/${REMOTE_PORT}/p2p/${remoteLibp2p.peerId.toString()}`))
+    expect(connection).to.exist()
+
+    // wait for identify to run on the new connection
+    await waitForIdentify(identityServiceIdentifySpy, connection, remoteLibp2p)
+
+    // assert we have received certified announce addresses
+    const peer = await libp2p.peerStore.get(remoteLibp2p.peerId)
+    expect(peer.addresses).to.have.lengthOf(1)
+    expect(peer.addresses[0].isCertified).to.be.true('did not receive certified address via identify')
+    expect(peer.addresses[0].multiaddr.toString()).to.startWith('/dns4/localhost/', 'did not receive announce address via identify')
+  })
+
+  it('should run identify automatically for inbound connections', async () => {
+    await libp2p.start()
+    await remoteLibp2p.start()
+
+    if (libp2p.identifyService == null) {
+      throw new Error('Identity service was not configured')
+    }
+
+    const identityServiceIdentifySpy = sinon.spy(libp2p.identifyService, 'identify')
+
+    // dial remote -> local via loopback in order to assert we receive the announce address via identify
+    const connection = await remoteLibp2p.dial(multiaddr(`/ip4/127.0.0.1/tcp/${LOCAL_PORT}/p2p/${libp2p.peerId.toString()}`))
+    expect(connection).to.exist()
+
+    // wait for identify to run on the new connection
+    await waitForIdentify(identityServiceIdentifySpy, connection, remoteLibp2p)
+
+    // assert we have received certified announce addresses
+    const peer = await libp2p.peerStore.get(remoteLibp2p.peerId)
+    expect(peer.addresses).to.have.lengthOf(1)
+    expect(peer.addresses[0].isCertified).to.be.true('did not receive certified address via identify')
+    expect(peer.addresses[0].multiaddr.toString()).to.startWith('/dns4/localhost/', 'did not receive announce address via identify')
+  })
+
+  it('should identify connection on dial and get proper announce addresses', async () => {
+    const announceAddrs = [
+      '/dns4/peers1.com/tcp/433/wss',
+      '/dns4/peers2.com/tcp/433/wss'
+    ]
+    const port = 58322
+    const protocol = '/ipfs/bitswap/1.2.0'
+
+    const receiver = await createNode({
+      config: {
+        addresses: {
+          announce: announceAddrs,
+          listen: [`/ip4/127.0.0.1/tcp/${port}/ws`]
+        }
+      }
+    })
+    await receiver.handle(protocol, () => {})
+
+    const sender = await createNode({
+      config: {
+        addresses: {
+          listen: ['/ip4/127.0.0.1/tcp/0/ws']
+        }
+      }
+    })
+
+    const connection = await sender.dial(multiaddr(`/ip4/127.0.0.1/tcp/${port}/ws/p2p/${receiver.peerId.toString()}`))
+
+    await delay(1000)
```

Comment:
what are we waiting for here? Can we use some spy instead?

Reply:
We are waiting for identify.  Since 0.45.x we can wait for the `peer:identify` event instead though.  Have updated.